vars/MavenCommand.groovy

```diff
@@ -0,0 +1,127 @@
+def class MavenCommand {
+
+    def steps
+
+    boolean returnStdout = false
+
+    String settingsXmlPath = ''
+    List mavenOptions = []
+    Map properties = [:]
+    String logFileName = null
+    List profiles = []
+
+    MavenCommand(steps){
+        this.steps = steps
+    }
+
+    MavenCommand(steps, List defaultOpts){
+        this(steps)
+        this.mavenOptions.addAll(defaultOpts)
+    }
+
+    def run(String goals) {
+        def cmd = "mvn -B"
+        if(this.settingsXmlPath) {
+            cmd += " -s ${this.settingsXmlPath}"
+        }
+        if(this.mavenOptions.size() > 0){
+            cmd += " ${this.mavenOptions.join(' ')}"
+        }
+        cmd += " ${goals}"
+        if(this.profiles.size() > 0){
+            cmd += " -P${this.profiles.join(',')}"
+        }
+        if(this.properties.size()){
+            cmd += " ${this.properties.collect{ it.value != '' ? "-D${it.key}=${it.value}" : "-D${it.key}" }.join(' ')}"
+        }
+        if(this.logFileName){
+            cmd += " | tee \$WORKSPACE/${logFileName} ; test \${PIPESTATUS[0]} -eq 0"
+        }
+
+        if(this.returnStdout){
+            return steps.sh(script: cmd, returnStdout: this.returnStdout)
+        }
+        steps.sh cmd
+    }
+
+    MavenCommand withSettingsXmlId(String settingsXmlId){
+        steps.configFileProvider([steps.configFile(fileId: settingsXmlId, variable: 'MAVEN_SETTINGS_XML')]) {
+            withSettingsXmlFile(steps.env['MAVEN_SETTINGS_XML'])
+        }
+        return this
+    }
+
+    MavenCommand withSettingsXmlFile(String settingsXmlPath){
+        if(!settingsXmlPath){
```

Comment:
I would replace this by
`assert settingsXmlPath: 'Trying to set an empty settings xml path'`